src/components/Map/map.js

```diff
@@ -177,8 +177,16 @@ class Map extends Component {
           visibility: "visible"
         }
       });
-      this.addClickHandlerToMapIdLayer(typeId);
-      this.addHoverHandlerToMapIdLayer(typeId);
+      this.map.on("click", typeId, e => this.providerLayerClickHandler(e));
```

Comment:
IIRC you refactored addClick/Hover/ClusterClick so the marker could show popups and handle click events. Since we're now using the marker purely for visual effect, do you think it still makes sense to refactor these methods (does it make cluster highlighting easier)? If not then I think this file only needs the new markRecentSelection method. If you want to keep the refactor I think you need to add clusterClickHandler to findClustersInMap (clicking a cluster doesn't work)

Reply:
Yes, I do think we should keep this refactor in general. When I extracted these originally, I really overcomplicated things, and this refactor just basically unpacks that.
True, though, this isn't relevant to this PR anymore, so I'll remove. 

(Good catch on clusterClickHandler, that goes back into findClustersInMap either way.)